### PR TITLE
Move the Payment booster view logic

### PR DESCRIPTION
**Pull request by pavel-bystritskyi**

### CODE REVIEW

**Comment by victor-petryk** [comment]
I guess, we need to update comment on configuration section, as this standalone module will not have these options without additional modules.

https://github.com/bold-commerce/adobe-commerce-bold-checkout/blob/ed8caab2a4a047427e7290992b57b2826cc3fac2/etc/adminhtml/system.xml#L24

https://github.com/bold-commerce/adobe-commerce-bold-checkout/blob/ed8caab2a4a047427e7290992b57b2826cc3fac2/etc/adminhtml/system.xml#L25

File: Model/RedirectToBoldCheckout/IsOrderInitializationAllowedForCart.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsOrderInitializationAllowedForCart.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsOrderInitializationAllowedPool.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsOrderInitializationAllowedPool.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedByParallel.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedByParallel.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedByParallel.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     *
+     * @param CartInterface $quote
+     * @param RequestInterface $request
+     * @return bool
+     */
+    public function isAllowed(CartInterface $quote, RequestInterface $request): bool
+    {
+        $websiteId = (int)$quote->getStore()->getWebsiteId();
+        if ($this->config->isCheckoutTypeParallel($websiteId)) {
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedByParallel.php:40]: Redundant row

File: Model/RedirectToBoldCheckout/IsOrderInitializationAllowedForRequest.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsOrderInitializationAllowedForRequest.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsOrderInitializationAllowedInterface.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsOrderInitializationAllowedInterface.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsOrderInitializationAllowedPool.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+use Magento\Framework\App\RequestInterface;
+use Magento\Quote\Api\Data\CartInterface;
+
+/**
+ * Pool for Bold order initialization is availability checkers.
+ */
+class IsOrderInitializationAllowedPool implements IsOrderInitializationAllowedInterface
+{
+    /**
+     * @var array
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsOrderInitializationAllowedPool.php:14]: `IsOrderInitializationAllowedInterface[]` ?

File: Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedInterface.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedInterface.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedPool.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
+<?php
+
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedPool.php:2]: `declare(strict_types=1);`

File: Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedPool.php [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+use Magento\Framework\App\RequestInterface;
+use Magento\Quote\Api\Data\CartInterface;
+
+/**
+ * Check if the redirect to Bold checkout allowed pool.
+ */
+class IsRedirectToBoldCheckoutAllowedPool implements IsRedirectToBoldCheckoutAllowedInterface
+{
+    private $elementList;
```
Comment (victor-petryk) [inline Model/RedirectToBoldCheckout/IsRedirectToBoldCheckoutAllowedPool.php:13]: docblock

File: Observer/Checkout/RedirectToBoldCheckoutObserver.php [hunk 11]
```diff
@@ -55,30 +45,45 @@ class RedirectToBoldCheckoutObserver implements ObserverInterface
...
+
+    /**
+     * @var IsRedirectToBoldCheckoutAllowedInterface
+     */
+    private $isRedirectToBoldCheckoutAllowed;
+
+    /**
+     * @var array
+     */
+    private $test;
```
Comment (victor-petryk) [inline Observer/Checkout/RedirectToBoldCheckoutObserver.php:60]: redundant

File: Observer/Order/SaveOrderExtensionDataObserver.php [hunk 12]
```diff
@@ -76,10 +74,6 @@ public function execute(Observer $observer)
         $orderExtensionData = $this->orderExtensionDataFactory->create();
         $orderExtensionData->setOrderId($orderId);
         $orderExtensionData->setPublicId($publicOrderId);
-        $this->eventManager->dispatch(
```
Comment (victor-petryk) [inline Observer/Order/SaveOrderExtensionDataObserver.php:79]: As I know this event used before to save additional data to order, like data from LiFE elements. Any reason why we need to remove this?

File: etc/db_schema.xml [hunk 13]
```diff
@@ -22,4 +22,15 @@
             <column name="quote_id"/>
         </index>
     </table>
+    <table name="bold_checkout_quote" resource="default" engine="innodb" comment="Is Bold Checkout Order In Progress">
```
Comment (victor-petryk) [inline etc/db_schema.xml:25]: comment ?

File: Model/Order/InitOrderFromQuote/ProcessBoldOrderData.php [hunk 14]
```diff
@@ -0,0 +1,71 @@
+<?php
```
Comment (victor-petryk) [inline Model/Order/InitOrderFromQuote/ProcessBoldOrderData.php:1]: It seems its not needed for bold hosted checkout

File: etc/frontend/di.xml [hunk 15]
```diff
@@ -1,15 +1,30 @@
...
+    </type>
+    <type name="Bold\Checkout\Model\RedirectToBoldCheckout\IsRedirectToBoldCheckoutAllowedPool">
+        <arguments>
+            <argument name="elementList" xsi:type="array">
+                <item name="isAllowedByParallelCheckout" xsi:type="object">Bold\Checkout\Model\RedirectToBoldCheckout\IsRedirectToBoldCheckoutAllowedByParallel</item>
+            </argument>
         </arguments>
     </type>
-    <type name="Magento\Checkout\Model\CompositeConfigProvider">
+    <type name="Bold\Checkout\Model\Order\InitOrderFromQuote">
```
Comment (victor-petryk) [inline etc/frontend/di.xml:24]: no processors needed for bold hosted checkout at this point

**Review by victor-petryk — APPROVED** [review]

**Review by victor-petryk — APPROVED** [review]

**Review by victor-petryk — APPROVED** [review]

**Review by TimeBones — APPROVED** [review]

**Pull request merged**